### PR TITLE
SSL Lhotse Dataloader Support

### PR DESCRIPTION
**Pull request by monica-sekoyan**
# What does this PR do ?

Added Lhotse Dataloader to `EncDecSpeechDenoiseMLMModel`

**Collection**: [Note which collection this PR will affect]

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [x] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

**Review by nithinraok — COMMENTED** [review]
LGTM

File: workspace/train_tarred_dns_lhotse.sh [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    model.train_ds.num_workers=$num_workers \
+    model.validation_ds.num_workers=$num_workers \
+    ++trainer.gradient_clip_val=1.0 \
+    ++trainer.val_check_interval=1.0 \
+    ++trainer.max_epochs=10 \
+    exp_manager.name=$exp_name \
+    exp_manager.create_wandb_logger=False \
+    exp_manager.wandb_logger_kwargs.name=$exp_name \
+    exp_manager.wandb_logger_kwargs.project="ssl_WavLM" \
+    ++model.train_ds.batch_duration=120 \
```
Comment (pzelasko) [inline workspace/train_tarred_dns_lhotse.sh:43]: I strongly suggest running `python scripts/speech_recognition/estimate_duration_bins.py` with your data. It will output an extra option to add here `++model.train_ds.duration_bins=[...]` which will optimize your bucketing sampling and speed up the start of the training.

Comment (pzelasko) [reply workspace/train_tarred_dns_lhotse.sh:43]: Also your batch duration is pretty low. You can probably increase it quite a lot (monitor nvidia-smi / W&B GPU memory usage and aim for as close to 100% as possible). Soon I'm going to merge a batch size OOMptimizer PR https://github.com/NVIDIA/NeMo/pull/9763 which can also support SSL models for getting 100% GPU utilization and much faster training.
